### DISCUSSION TITLE
how to add  webviews when JavaScript calls a command?

### DISCUSSION THREAD

**Discussion by cathino** [tauri-apps/tauri#10236] [Q&A] [answered]
I found in the Tauri example (https://github.com/tauri-apps/tauri/blob/dev/examples/multiwebview/main.rs) that webviews can be added in the **setup** function. 

Is there any way to do this **dynamically when JavaScript calls a command**?

Thank you.

**Comment by FabianLars** [comment]
Yes, you can call the WebviewBuilder in a normal (async) [command](https://v2.tauri.app/develop/calling-rust/#accessing-an-apphandle-in-commands) just fine.

**Reply by cathino** [reply]
I'd like to use some code like:

```rust
#[tauri::command]
pub fn add_webview(window: tauri::Window, app: tauri::App) {
    let window = tauri::window::WindowBuilder::new(&app, "main")
        .inner_size(width, height)
        .build()?;

    let _webview1 = window.add_child(
        tauri::webview::WebviewBuilder::new("main1", WebviewUrl::App(Default::default()))
            .auto_resize(),
        LogicalPosition::new(0., 0.),
        LogicalSize::new(width / 2., height / 2.),
    )?;
}
```

But there's an error:

```
struct `WindowBuilder` is private
private structrustcClick for full compiler diagnostic
mod.rs(108, 7): the struct `WindowBuilder` is defined here
```

**Reply by FabianLars** [reply, marked as answer]
Multi webview support is currently wip/unstable so you have to add the `"unstable"` feature flag to the tauri dependency in Cargo.toml.
Then the apis should not be private anymore.

**Reply by cathino** [reply]
Thank you, I made it.
